main.go

```diff
@@ -72,67 +72,8 @@ type CacheModel struct {
 	IndicatorHashMap    map[string]string
 }
 
-// ConfigsModel ...
-type ConfigsModel struct {
-	Paths               string
-	IgnoredPaths        string
-	DebugMode           string
-	CacheAPIURL         string
-	FingerprintMethodID string
-	CompressArchive     string
-}
-
-func (configs *ConfigsModel) print() {
-	log.Printf("- CachePaths:")
-	for _, path := range strings.Split(configs.Paths, "\n") {
-		clnPth := strings.TrimSpace(path)
-		if clnPth == "" {
-			continue
-		}
-		log.Printf("  * %s", clnPth)
-	}
-
-	fmt.Println()
-
-	log.Printf("- IgnoredPaths:")
-	for _, path := range strings.Split(configs.IgnoredPaths, "\n") {
-		clnPth := strings.TrimSpace(path)
-		if clnPth == "" {
-			continue
-		}
-		log.Printf("  * %s", clnPth)
-	}
-
-	fmt.Println()
-
-	log.Printf("- CompressArchive: %s", configs.CompressArchive)
-	log.Printf("- FingerprintMethodID: %s", configs.FingerprintMethodID)
-}
-
-func (configs *ConfigsModel) validate() error {
-	if err := input.ValidateIfNotEmpty(configs.CacheAPIURL); err != nil {
-		return fmt.Errorf("CacheAPIURL: %s", err)
-	}
-	if err := input.ValidateWithOptions(configs.FingerprintMethodID, fingerprintMethodIDContentChecksum, fingerprintMethodIDFileModTime); err != nil {
-		return fmt.Errorf("FingerprintMethodID: %s", err)
-	}
-
-	return nil
-}
-
-func createConfigsModelFromEnvs() *ConfigsModel {
-	return &ConfigsModel{
-		DebugMode:           os.Getenv("is_debug_mode"),
-		CacheAPIURL:         os.Getenv("cache_api_url"),
-		CompressArchive:     os.Getenv("compress_archive"),
-		FingerprintMethodID: os.Getenv("fingerprint_method"),
-		Paths:               os.Getenv("cache_paths") + "\n" + os.Getenv("bitrise_cache_include_paths"),
-		IgnoredPaths:        os.Getenv("ignore_check_on_paths") + "\n" + os.Getenv("bitrise_cache_exclude_paths"),
-	}
-}
-
 // NewCacheModel ...
-func NewCacheModel(configs *ConfigsModel) *CacheModel {
+func NewCacheModel(configs *Config) *CacheModel {
```

Comment:
Why do we use pointer here (*Config), instead of the default copy?
This method will not change anything in the configs.